src/store/accounts/nonce.ts

```diff
@@ -21,11 +23,22 @@ export const latestNonceState = atom(get => {
   return get(localNoncesState([network.url, address || '']));
 });
 
+export const lastApiNonce = atom(async get => {
+  const principal = get(currentAccountStxAddressState);
+  if (!principal) return;
+
+  const { accountsApi } = get(apiClientState);
+  return accountsApi.getAccountNonces({ principal });
+});
+
 export const correctNonceState = atomFamily<string, number>(address =>
   atom(get => {
     const account = get(accountInfoState);
     const accountData = get(currentAccountDataState);
     const lastLocalNonce = get(latestNonceState);
+    const apiNonce = get(lastApiNonce);
+    // We try to use the api nonce first since it will be the most accurate value
+    if (apiNonce && !isNaN(apiNonce.possible_next_nonce)) return +apiNonce.possible_next_nonce;
```

Comment:
This will basically always be true, right? The rest of this code will then never be of use. I think we should probably do some more checks

Reply:
Yes unless there is an api issue but if the api does not respond our wallet is not resilient to that anyway. 
I am not sure what other checks I can do here. What am I missing ?

Reply:
I think two things:
- this value can be cached, and potentially out of date, in that situation based on how this is written, the only value that will ever return is `+apiNonce.possible_next_nonce`. and would never use the local nonce we set based on user activity.
- why are we incrementing this `possible_next_nonce`? (does the `+` in front of it indicate that?) to me it seems like it's already the _next_ nonce.

Reply:
1. Got it, then maybe I should use our new async way to get the latest ? Or is there anyway I can 'force' jotai to get the latest every time (in this case we always want the latest).
2. It's not an increment but a cast to number